pkg/db/db_sync_status.go

```diff
@@ -0,0 +1,244 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright freiheit.com*/
+
+package db
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/freiheit-com/kuberpult/pkg/logger"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"time"
+)
+
+type SyncStatus int
+
+const (
+	SYNCED = iota
+	UNSYNCED
+	SYNC_FAILED
+)
+
+const BULK_INSERT_BATCH_SIZE = 500
+
+type GitSyncData struct {
+	AppName       string
+	EnvName       string
+	TransformerID EslVersion
+	SyncStatus
+}
+
+func (h *DBHandler) DBWriteNewSyncEvent(ctx context.Context, tx *sql.Tx, syncData *GitSyncData) error {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBWriteNewSyncEvent")
+	defer span.Finish()
+	if h == nil {
+		return nil
+	}
+	if tx == nil {
+		return fmt.Errorf("DBWriteNewSyncEvent: no transaction provided")
+	}
+
+	insertQuery := h.AdaptQuery("INSERT INTO git_sync_status (created, transformerid, envName, appName, status)  VALUES (?, ?, ?, ?, ?);")
+	now, err := h.DBReadTransactionTimestamp(ctx, tx)
+	if err != nil {
+		return fmt.Errorf("DBWriteNewSyncEvent unable to get transaction timestamp: %w", err)
+	}
+	span.SetTag("query", insertQuery)
+	_, err = tx.Exec(
+		insertQuery,
+		*now,
+		syncData.TransformerID,
+		syncData.EnvName,
+		syncData.AppName,
+		syncData.SyncStatus)
+
+	if err != nil {
+		return fmt.Errorf("could not write sync event into DB. Error: %w\n", err)
+	}
+	return nil
+}
+
+func (h *DBHandler) DBWriteNewSyncEventBulk(ctx context.Context, tx *sql.Tx, id TransformerID, envApps []EnvApp, status SyncStatus) error {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBWriteNewSyncEventBulk")
+	defer span.Finish()
+	if h == nil {
+		return nil
+	}
+	if tx == nil {
+		return fmt.Errorf("DBWriteNewSyncEventBulk: no transaction provided")
+	}
+	now, err := h.DBReadTransactionTimestamp(ctx, tx)
+	if err != nil {
+		return fmt.Errorf("DBWriteNewSyncEventBulk unable to get transaction timestamp: %w", err)
+	}
+
+	err = h.executeBulkInsert(ctx, tx, envApps, *now, id, status, BULK_INSERT_BATCH_SIZE)
+
+	if err != nil {
+		return fmt.Errorf("could not write sync event into DB. Error: %w\n", err)
+	}
+	return nil
+}
+
+type EnvApp struct {
+	AppName string
+	EnvName string
+}
+
+func (h *DBHandler) DBReadUnsyncedAppsForTransfomerID(ctx context.Context, tx *sql.Tx, id TransformerID) ([]EnvApp, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "DBReadUnsyncedAppsForTransfomerID")
```

Comment:
please use the function `tracing.StartSpanFromContext`, so the span is marked as error

Reply:
(in all functions that return an error)